File: lib/core/models/ble_connect_source.dart
/// 一次连接状态事件的触发来源。
///
/// `unknown` 保留给旧版原生 payload 和普通首连，避免新增字段破坏历史调用方。
enum BleConnectSource {
  unknown,
  autoReconnect,
  manualReconnect,
  stateRestoration,
  foreground,
}

File: android/src/main/kotlin/com/fzfstudio/ezw_ble/ble/models/BleConnectSource.kt
package com.fzfstudio.ezw_ble.ble.models

import com.google.gson.annotations.SerializedName

/** 与 Dart `BleConnectSource` raw value 一致的原生连接来源。 */
enum class BleConnectSource(val flutterValue: String) {
    @SerializedName("unknown")
    UNKNOWN("unknown"),
    @SerializedName("autoReconnect")
    AUTO_RECONNECT("autoReconnect"),
    @SerializedName("manualReconnect")
    MANUAL_RECONNECT("manualReconnect"),
    @SerializedName("stateRestoration")
    STATE_RESTORATION("stateRestoration"),
    @SerializedName("foreground")
    FOREGROUND("foreground");

    companion object {
        /** 未知未来值必须降级为 UNKNOWN，保证跨版本 EventChannel 兼容。 */
        fun fromFlutterValue(value: String?): BleConnectSource =
            entries.firstOrNull { it.flutterValue == value } ?: UNKNOWN
    }
}
